File: Go_leftToDo/lists/taskList.go
package lists

import (
	"fmt"
	"strconv"

	"github.com/WiviWonderWoman/DevUx/Go/tasks"
)

type TaskList struct {
	ToDoList []tasks.Task
	Archive  []tasks.Task
}

func NewTaskList() TaskList {
	return TaskList{
		ToDoList: []tasks.Task{},
		Archive:  []tasks.Task{},
	}
}

func (tl *TaskList) AddToDoTask(task tasks.Task) {
	tl.ToDoList = append(tl.ToDoList, task)
}

func (tl *TaskList) addTaskToArhive(task tasks.Task) {
	tl.Archive = append(tl.Archive, task)
}

func (tl *TaskList) ArchiveTask() {
	tdl := []tasks.Task{}
	for i := 0; i < len(tl.ToDoList); i++ {
		task := tl.ToDoList[i]
		if task.Done == true {
			tl.addTaskToArhive(tl.ToDoList[i])
		} else if task.Done == false {
			tdl = append(tdl, tl.ToDoList[i])
		}
	}
	tl.ToDoList = append(tdl)
}

func (tl *TaskList) ShowLeftToDo() {
	if len(tl.ToDoList) < 1 {
		fmt.Println("\t\tATT GÖRA LISTAN ÄR TOM!")
		return
	}
	fmt.Println("Status\tNr.\tUppgift")

	for index, task := range tl.ToDoList {
		task.ShowTask(index)
	}
}

func (tl *TaskList) FindTaskToMark() {
	fmt.Printf("\n\nVilken uppgift vill du markera / avmarkera? Är det en under uppgift, ange först rubrikens nummer.\n\n")
	input := readInt()
	for i := 0; i < len(tl.ToDoList); i++ {
		task := tl.ToDoList[i]
		if input == i {
			if task.TaskType == "S" {
				tl.ToDoList[i] = task.MarkAsDone(task)
			} else if task.TaskType == "C" {
				fmt.Printf("\n\nVilken underuppgift vill du markera / avmarkera?\n\n")
				subInput := readInt()
				count := len(task.SubTask)
				marked := 0
				for j := 0; j < count; j++ {
					subTask := task.SubTask[j]
					if subTask.Done {
						marked++
					}
					if subInput == j {
						*task.SubTask[j] = subTask.CreateMarkAsDone(*subTask)
						subTask.Done = !subTask.Done
						marked++
					}
				}
				if count == marked {
					tl.ToDoList[i] = task.MarkAsDone(task)
				}
			}
		}
	}
	tl.ShowLeftToDo()
}

func readInt() int {

	var input string
	fmt.Scanln(&input)
	number, err := strconv.Atoi(input)
	for err != nil {
		fmt.Printf("\n\t\tDu skrev inte in en siffra. Försök igen.\n\n")
		fmt.Scanln(&input)
		number, err = strconv.Atoi(input)
	}
	return number - 1
}

func (tl *TaskList) ShowArchive() {
	if len(tl.Archive) < 1 {
		fmt.Printf("\n\t\tARKIVET ÄR TOMT.\n\n")
		return
	}
	fmt.Println("UTFÖRDA UPPGIFTER:\nStatus\tArkiv.\tUppgift")
	amount := 0
	for i := 0; i < len(tl.Archive); i++ {
		task := tl.Archive[i]
		amount++
		task.ShowTask(i + 1)
		if task.TaskType == "C" {
			for j := 0; j < len(task.SubTask); j++ {
				sub := task.SubTask[j]
				sub.ShowSubTask(i+1, j+1)
			}
		}
	}
	fmt.Printf("\n\t\tWOW! DU HAR UTFÖRT %v UPPGIFTER!\n", amount)
}

File: C#_LeftToDo/src/Lists/TaskList.cs
using System;
using System.Collections.Generic;
using LeftToDo.Tasks;

namespace LeftToDo.Lists
{
    /* ToDolist a class that handles all lists and their methods */
    public class TaskList
    {
        public List<Task> ToDoList // Tasks ToDo
        {
            get;
            private set;
        }
        public List<Task> Archive // Archived done Tasks
        {
            get;
            private set;
        }

        // Constructor
        public TaskList()
        {
            ToDoList = new List<Task>();
            Archive = new List<Task>();
        }

        // Adds new Task to ToDoList
        public void AddToDoTask(Task task)
        {
            ToDoList.Add(task);
        }

        // Adds Task to Archive
        private void AddTaskToArhive(Task task)
        {
            Archive.Add(task);
        }

        // Iterates thru ToDoList to find done Task to archive
        public void ArchiveTask()
        {
            for (int i = 0; i < ToDoList.Count; i++)
            {
                var task = ToDoList[i];
                if (task.done)
                {
                    AddTaskToArhive(task);
                    ToDoList.Remove(task);
                    i--;
                }
            }
        }

        /// <summary>Display ToDoList</summary>  
        internal void ShowLeftToDo(List<Task> ToDoList)
        {
            if (ToDoList.Count < 1)
            {
                Console.WriteLine($"\t\tATT GÖRA LISTAN ÄR TOM!");
                return;
            }
            Console.WriteLine($"Status\tNr.\tUppgift\n");

            int index = 1;
            foreach (var task in ToDoList)
            {
                Task.ShowTask(task, index);
                index++;
            }
        }

        // Find Task to mark as done / undone
        internal void FindTaskToMark(TaskList list)
        {
            Console.WriteLine("\nVilken uppgift vill du markera / avmarkera? Är det en under uppgift, ange först rubrikens nummer.\n");
            int index = ReadInt() - 1;

            for (int i = 0; i < list.ToDoList.Count; i++)
            {
                var task = list.ToDoList[i];
                if (i == index)
                {
                    if (task.type == "C")
                    {
                        Console.WriteLine("\nVilken underuppgift vill du markera / avmarkera?\n");
                        int subIndex = ReadInt() - 1;

                        var count = task.subTask.Count;
                        var marked = 0;

                        for (int j = 0; j < count; j++)
                        {
                            var subTask = task.subTask[j];
                            if (subTask.done)
                            {
                                marked++;
                            }
                            if (j == subIndex)
                            {
                                subTask.MarkAsDone();
                                marked++;
                            }
                        }
                        if (count == marked)
                        {
                            task.MarkAsDone();
                        }
                    }
                    else
                    {
                        task.MarkAsDone();
                    }
                }
            }
            list.ShowLeftToDo(list.ToDoList);
        }

        // Parse input to an integer or displays error message
        private static int ReadInt()
        {
            int number;
            while (int.TryParse(Console.ReadLine(), out number) == false)
            {
                Console.WriteLine("\t\tDu skrev inte in en siffra. Försök igen.\n\n");
            }
            return number;
        }

        // Display Archive
        internal void ShowArchive(List<Task> Archive)
        {
            if (Archive.Count < 1)
            {
                Console.WriteLine($"\t\tARKIVET ÄR TOMT.\n\n");
                return;
            }

            Console.WriteLine($"UTFÖRDA UPPGIFTER:\nStatus\tArkiv.\tUppgift\n");
            int amount = 0;

            for (int i = 0; i < Archive.Count; i++)
            {
                var task = Archive[i];
                amount++;
                if (task.type == "C")
                {
                    Checklist.ShowTask(task, i + 1);
                    for (int j = 0; j < task.subTask.Count; j++)
                    {
                        var sub = task.subTask[j];
                        SimpleTask.ShowTask(sub, i + 1, j + 1);
                        amount++;
                    }
                }
                else if (task.type == "S")
                {
                    SimpleTask.ShowTask(task, i + 1);
                }
            }
            Console.WriteLine($"\t\tWOW! DU HAR UTFÖRT {amount} UPPGIFTER!\n\n");
        }
    }
}
